ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-isis.j2

```diff
@@ -34,30 +34,51 @@
+{%              if vlan_interfaces[vlan_interface].isis_enable is arista.avd.defined %}
+| {{ vlan_interface }} | {{ vlan_interfaces[vlan_interface].isis_enable }} |  {{ vlan_interfaces[vlan_interface].isis_metric | default('-') }} | {% if vlan_interfaces[vlan_interface].isis_network_point_to_point is arista.avd.defined %} point-to-point {% elif vlan_interfaces[vlan_interface].isis_passive is arista.avd.defined %} passive {% else %} - {% endif %}|
 {%              endif %}
 {%          endfor %}
 {%      endif %}
 {# ------- #}
 {# loopback interfaces #}
 {# ------- #}
-{%      if loopback_interfaces is defined and loopback_interfaces is not none %}
+{%      if loopback_interfaces is arista.avd.defined %}
 {%          for loopback_interface in loopback_interfaces | arista.avd.natural_sort %}
-{%              if loopback_interfaces[loopback_interface].isis_enable is defined and loopback_interfaces[loopback_interface].isis_enable is not none %}
-| {{ loopback_interface }} | {{ loopback_interfaces[loopback_interface].isis_enable }} |  {{ loopback_interfaces[loopback_interface].isis_metric | default('-') }} | {% if loopback_interfaces[loopback_interface].isis_network_point_to_point is defined and loopback_interfaces[loopback_interface].isis_network_point_to_point is not none %} point-to-point {% elif loopback_interfaces[loopback_interface].isis_passive is defined and loopback_interfaces[loopback_interface].isis_passive is not none %} passive {% else %} - {% endif %}|
+{%              if loopback_interfaces[loopback_interface].isis_enable is arista.avd.defined %}
+| {{ loopback_interface }} | {{ loopback_interfaces[loopback_interface].isis_enable }} |  {{ loopback_interfaces[loopback_interface].isis_metric | default('-') }} | {% if loopback_interfaces[loopback_interface].isis_network_point_to_point is arista.avd.defined %} point-to-point {% elif loopback_interfaces[loopback_interface].isis_passive is arista.avd.defined %} passive {% else %} - {% endif %}|
+{%              endif %}
+{%          endfor %}
+{%      endif %}
+{# ISIS Node-SID #}
+{%      set node_sid = namespace() %}
+{%      for loopback_interface in loopback_interfaces | arista.avd.natural_sort %}
+{%          if loopback_interfaces[loopback_interface].node_segment.ipv4_index is arista.avd.defined or loopback_interfaces[loopback_interface].node_segment.ipv6_index is arista.avd.defined %}
+{%              set node_sid.configured = true %}
+{%          endif %}
+{%      endfor %}
+{%      if node_sid.configured is arista.avd.defined(true) %}
+
+### ISIS Segment-routing Node-SID
+
+| Loopback | IPv4 Index | IPv6 Index |
+| -------- | ---------- | ---------- |
+{%          for loopback_interface in loopback_interfaces | arista.avd.natural_sort %}
+{%              if loopback_interfaces[loopback_interface].node_segment.ipv4_index is arista.avd.defined or loopback_interfaces[loopback_interface].node_segment.ipv6_index is arista.avd.defined %}
```

Comment:
We don't need this check once again if you follow the above coding technique.

Reply:
Fixed.